sdk/azcore/runtime/request.go

```diff
@@ -94,7 +95,9 @@ func MarshalAsByteArray(req *policy.Request, v []byte, format Base64Encoding) er
 
 // MarshalAsJSON calls json.Marshal() to get the JSON encoding of v then calls SetBody.
 func MarshalAsJSON(req *policy.Request, v interface{}) error {
-	v = cloneWithoutReadOnlyFields(v)
+	if omit := os.Getenv("AZURE_SDK_GO_OMIT_READONLY"); omit != "" {
```

Comment:
I'm assuming this is related to a conversation that happeened in the arch board.

Are we sure this should be a global setting, and not one that's passed in so we can change it for each service, if necessary?

Reply:
@JeffreyRichter and I debated this approach. For large code bases, it could be expensive for customers to toggle the setting at the client/API level. There was also some discussion with the ARM team about our approach and based on the provided [docs](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html) we think this is an acceptable path.

Reply:
If we did ever decide to make it configurable, per-client, would we say that the newly added setting (in code, not the environment variable) would override the env variable's value?

In a similar vein, do we think people will set AZURE_SDK_GO_OMIT_READONLY='no' or 'false' and expect it to change the meaning?

Reply:
We really want to never make this configurable. It's just an escape hatch if a service has incorrect behavior.

Reply:
Based on yoiur reply to @chlowell, I'm good. Adding some strictness around it would be good, and removing it at some point (which sounds like our plan) is even better.